platforms/software/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/BuildInitInteractiveIntegrationTest.groovy

```diff
@@ -119,48 +129,50 @@ class BuildInitInteractiveIntegrationTest extends AbstractInitIntegrationSpec {
 
         // Select 'application'
         ConcurrentTestUtil.poll(60) {
-            assert handle.standardOutput.contains(projectTypePrompt)
+            assert handle.standardOutput.contains(buildTypePrompt)
         }
-        handle.stdinPipe.write(("2" + TextUtil.platformLineSeparator).bytes)
+        handle.stdinPipe.write((applicationOption + TextUtil.platformLineSeparator).bytes)
 
         // Select 'java'
         ConcurrentTestUtil.poll(60) {
             languageSelectionOptions.each {
                 assert handle.standardOutput.contains(it)
             }
         }
-        handle.stdinPipe.write(("3" + TextUtil.platformLineSeparator).bytes)
+        handle.stdinPipe.write((javaOption + TextUtil.platformLineSeparator).bytes)
 
-        // Select 'Single project'
+        // Enter a Java version
         ConcurrentTestUtil.poll(60) {
-            assert handle.standardOutput.contains("Generate multiple subprojects for application?")
+            assert handle.standardOutput.contains("Enter target Java version (min: 7, default: 21)")
         }
-        handle.stdinPipe.write(("no" + TextUtil.platformLineSeparator).bytes)
+        handle.stdinPipe.write(("17" + TextUtil.platformLineSeparator).bytes)
 
-        // Select 'kotlin' DSL
+        // Select default project name
         ConcurrentTestUtil.poll(60) {
-            assert handle.standardOutput.contains(dslPrompt)
+            assert handle.standardOutput.contains(projectNamePrompt)
         }
-        handle.stdinPipe.write(("1" + TextUtil.platformLineSeparator).bytes)
+        handle.stdinPipe.write(TextUtil.platformLineSeparator.bytes)
 
-        // Select 'junit'
+        // Select 'Single project'
         ConcurrentTestUtil.poll(60) {
-            assert handle.standardOutput.contains("Select test framework:")
-            assert handle.standardOutput.contains("1: JUnit 4")
+            assert handle.standardOutput.contains("Select application structure:")
+            assert handle.standardOutput.contains("1: Single application project")
+            assert handle.standardOutput.contains("2: Application and library project")
```

Comment:
The options for `Select application structure:` don't make sense to me. As a new user, I wouldn't know the difference between options 1 and 2.

Reply:
For this PR, I think the prompt just needs to be better than it was before ("Generate multiple subprojects for application? (default: no) [yes, no] "), which I think it is.

I'll see if I can tweak it to make it clearer. We also have the option to say "if you don't know, use the default".